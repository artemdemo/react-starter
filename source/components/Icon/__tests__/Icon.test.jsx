import React from 'react';
import renderer from 'react-test-renderer';
import Icon from '../Icon';

describe('<Icon>', () => {
    it('Simple render', () => {
        const tree = renderer.create(
            <Icon />,
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('with name', () => {
        const tree = renderer.create(
            <Icon name='fire' />,
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('with className', () => {
        const tree = renderer.create(
            <Icon className='some-class' />,
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('with style', () => {
        const tree = renderer.create(
            <Icon style={{position: 'absolute'}} />,
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
