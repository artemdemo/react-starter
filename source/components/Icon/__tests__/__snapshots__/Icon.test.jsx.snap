// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon> Simple render 1`] = `null`;

exports[`<Icon> with className 1`] = `null`;

exports[`<Icon> with name 1`] = `
<span
  className="fa fa-fire"
  style={null}
/>
`;

exports[`<Icon> with style 1`] = `null`;
